File: src/main/java/com/scalag/vulkan/compute/LayoutInfo.java
package com.scalag.vulkan.compute;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * @author MarconZet
 * Created 25.04.2020
 */
@RequiredArgsConstructor
@Getter
public class LayoutInfo {
    private final int
            set,
            binding,
            size;
}

File: src/main/scala/com/scalag/vulkan/compute/LayoutInfo.scala
package com.scalag.vulkan.compute

/** @author
  *   MarconZet Created 25.04.2020
  */
case class LayoutInfo(sets: Seq[LayoutSet])
case class LayoutSet(id: Int, bindings: Seq[Binding])
case class Binding(id: Int, size: Int)
